.github/workflows/generate_chapters.yml

```diff
@@ -43,76 +32,21 @@ jobs:
       uses: actions/checkout@v2
       with:
         fetch-depth: 0
-    - name: Update CSS version for modified CSS files
-      if: github.event_name == 'push'
-      env:
-        COMMIT_SHA: ${{ github.event.commits[0].id }}
-      run: |
-        echo "files changed in commit $COMMIT_SHA:"
-        git diff-tree --no-commit-id --name-only -r $COMMIT_SHA
-        echo "Incrementing any references to changed CSS files"
-        git diff-tree --diff-filter=AM --no-commit-id --name-only -r $COMMIT_SHA src/static/css | grep "\.css" | sed 's!.*/!!' | sed "s/\(.*\)/s\/\1?v=[0-9][0-9]*\/\1?v=$(date +%Y%m%d%H%M%S)\//" | sed -f - -i src/templates/base/*/*.html
-        git diff-tree --diff-filter=AM --no-commit-id --name-only -r $COMMIT_SHA src/static/css | grep "\.css" | sed 's!.*/!!' | sed "s/\(.*\)/s\/\1?v=[0-9][0-9]*\/\1?v=$(date +%Y%m%d%H%M%S)\//" | sed -f - -i src/templates/base.html
-        git diff
-    - name: Update JS version for modified CSS files
-      if: github.event_name == 'push'
-      env:
-        COMMIT_SHA: ${{ github.event.commits[0].id }}
-      run: |
-        echo "files changed in commit $COMMIT_SHA:"
-        git diff-tree --no-commit-id --name-only -r $COMMIT_SHA
-        echo "Incrementing any references to changed JS files"
-        git diff-tree --diff-filter=AM --no-commit-id --name-only -r $COMMIT_SHA src/static/js | grep "\.js" | sed 's!.*/!!' | sed "s/\(.*\)/s\/\1?v=[0-9][0-9]*\/\1?v=$(date +%Y%m%d%H%M%S)\//" | sed -f - -i src/templates/base/*/*.html
-        git diff-tree --diff-filter=AM --no-commit-id --name-only -r $COMMIT_SHA src/static/js | grep "\.js" | sed 's!.*/!!' | sed "s/\(.*\)/s\/\1?v=[0-9][0-9]*\/\1?v=$(date +%Y%m%d%H%M%S)\//" | sed -f - -i src/templates/base.html
-        git diff
-    - name: Update timestamp for modified files
-      if: github.event_name == 'push'
+    - name: Update version numbers and timestamps
       env:
         COMMIT_SHA: ${{ github.event.commits[0].id }}
```

Comment:
Done. Will do some more testing along with other changed below but will resolve that for now assuming it works.

Reply:
I would suggest you specially test it against the manual invocation and see what SHA is provided. Most likely, it will be the hash of the last commit on the default branch.

Reply:
I had to revert this back as it didn't work on merge.

Reply:
That's surprising. I think we can debug it in isolation somewhere else to see why it might not be working and them update the code later.

Reply:
Tell a lie - it does work. I'd just commented out part of my code to make test run faster - and then wondered why that didn't execute 😀